pkg/datastore/datastore.go

```diff
@@ -166,13 +206,49 @@ func (ds *memoryDatastore) RemoveOperatorSource(uid types.UID) {
 	delete(ds.rows, uid)
 }
 
-func (ds *memoryDatastore) GetOperatorSource(opsrcUID types.UID) (opsrc *v1alpha1.OperatorSourceSpec, ok bool) {
+func (ds *memoryDatastore) GetOperatorSource(opsrcUID types.UID) (*OperatorSourceKey, bool) {
 	row, exists := ds.rows[opsrcUID]
 	if !exists {
 		return nil, false
 	}
 
-	return row.Spec, true
+	return &row.OperatorSourceKey, true
+}
+
+func (ds *memoryDatastore) OperatorSourceHasUpdate(opsrcUID types.UID, metadata []*RegistryMetadata) (bool, error) {
+	// TODO: Return fine grained information that describes repository that
+	// was removed, added or has a new release.
+	row, exists := ds.rows[opsrcUID]
+	if !exists {
+		return false, fmt.Errorf("datastore has no record of the specified OperatorSource [%s]", opsrcUID)
+	}
+
+	// Right now we consider remote and local operator source to be same only
+	// when the following is true:
```

Comment:
`are true`

Reply:
On another note, it will be more useful to say what is the criteria for an `OperatorSource` to have an update.

Reply:
```
	//   New repositories added to the operator source.
	//   Repositories that have been removed from the operator source.
	//   Repositories that have a new version.
```
I already have it in the doc for `OperatorSourceHasUpdate` function. Is this what you were looking for?

Reply:
If it is there in the doc for that function, it can be repeated here. No reason of the reverse documentation.

Reply:
The comments I put inside the function is to explain the `for loop`. I prefer not to repeat the entire doc of the function but if you think repeating it is useful then I can copy it.

Reply:
I don't want it to be repeated. It just seems that you are restating the conditions in a negated fashion. Let me go through your next set of changes and see how it reads. I will comment if it needs any change.